analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssertionsShouldBeComplete.FluentAssertions.CSharp7.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using FluentAssertions.Primitives;
+
+public class Program
+{
+    public void StringAssertions()
+    {
+        var s = "Test";
+        s.Should();                   // Noncompliant {{Complete the assertion}}
+        //^^^^^^
+        s[0].Should();                // Error [CS0121] ambiguous calls
+                                      // Noncompliant@-1
+
+        s.Should().Be("Test");        // Compliant
+    }
+
+    public void CollectionAssertions()
+    {
+        var collection = new[] { "Test", "Test" };
+        collection.Should();                               // Error [CS0121] ambiguous calls
+                                                           // Noncompliant@-1
+        collection.Should<string>();                       // Noncompliant
+
+        collection.Should<string>().Equal("Test", "Test"); // Compliant
+    }
+
+    public void DictionaryAssertions()
+    {
+        var dict = new Dictionary<string, object>();
+        dict["A"].Should();    // Noncompliant
+    }
+
+    public StringAssertions ReturnedByReturn()
+    {
+        var s = "Test";
+        return s.Should();     // Compliant
+    }
+
+    public StringAssertions ReturnedByArrow(string s) =>
+        s.Should();            // Compliant
+
+    public object ReturnedByArrowWithConversion(string s) =>
+        (object)s.Should();    // Compliant
+
+    public void CalledByArrow(string s) =>
+        s.Should();            // Noncompliant
+
+    public void Assigned()
+    {
+        var s = "Test";
+        var assertion = s.Should();  // Compliant
+        assertion = s.Should();      // Compliant
+    }
+
+    public void PassedAsArgument()
+    {
+        var s = "Test";
+        ValidateString(s.Should());  // Compliant
+    }
+    private void ValidateString(StringAssertions assertion) { }
+
+    public void UnreducedCall()
+    {
+        var s = "Test";
+        FluentAssertions.AssertionExtensions.Should(s); // Noncompliant
+    }
+
+    public void ReturnedInLambda()
+    {
+        Func<StringAssertions> a = () => "Test".Should();
+    }
+
+    public void CustomAssertions()
+    {
+        var custom = new Custom();
+        custom.Should();                    // Noncompliant The custom assertion derives from ReferenceTypeAssertions
```

Comment:
We should also have a Compliant version that doesn't derive from ReferenceTypeAssertions to make sure we do not just check for the name.

Reply:
That is the CustomStructAssertions() test case.

Reply:
Seems I got more confused than I initially thought. Nvm